src/org/exist/Indexer.java

```diff
@@ -305,6 +305,9 @@ public void endDTD() {
     @Override
     public void endDocument() {
         if (!validate) {
+            if(indexListener != null) {
+                indexListener.endDocument(transaction);
```

Comment:
will `startDocument ` & `endDocument` be accepted here https://github.com/eXist-db/exist/blob/503ad3283e7f1cb7049b05eb3c5ed3d5d9a8b891/src/org/exist/dom/persistent/ElementImpl.java#L533 ? (as example)